bluegreen_deploy.py

```diff
@@ -65,228 +67,328 @@ def list_marathon_apps(args):
-            port = m.group(5)
-            if host in hostports:
-                hostports[host].append(int(port))
-            else:
-                hostports[host] = [int(port)]
-    return hostports
-
-
-def find_tasks_to_kill(tasks, hostports):
-    tasks_to_kill = set()
+def _get_alias_records(hostname):
+    """Return all IPv4 A records for a given hostname
+    """
+    return socket.gethostbyname_ex(hostname)[2]
+
+
+def _unparse_url_alias(url, addr):
+    """Reassemble a url object into a string but with a new address
+    """
+    return urllib.parse.urlunparse((url[0],
+                                    addr + ":" + str(url.port),
+                                    url[2],
+                                    url[3],
+                                    url[4],
+                                    url[5]))
+
+
+def get_marathon_lb_urls(args):
+    """Return a list of urls for all Aliases of the
+       marathon_lb url passed in as an argument
+    """
+    url = urllib.parse.urlparse(args.marathon_lb)
+    addrs = _get_alias_records(url.hostname)
+    return [ _unparse_url_alias(url, addr) for addr in addrs ]
+
+
+def fetch_haproxy_pids(haproxy_url):
+    try:
+        response = requests.get(haproxy_url + "/_haproxy_getpids")
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logger.exception("Caught exception when retrieving HAProxy"
+                         " pids from " + haproxy_url)
+        raise e
+
+    return response.text.split()
+
+
+def check_haproxy_reloading(haproxy_url):
+    """Return False if haproxy has only one pid, it is not reloading.
+       Return True if we catch an exception while making a request to
+       haproxy or if more than one pid is returned
+    """
+    try:
+        pids = fetch_haproxy_pids(haproxy_url)
+    except requests.exceptions.RequestException as e:
+        return True
+
+    if len(pids) > 1:
+        logger.info("Waiting for {} pids on {}".format(len(pids), haproxy_url))
+        return True
+
+    return False
+
+
+def any_marathon_lb_reloading(marathon_lb_urls):
+    return any([check_haproxy_reloading(url) for url in marathon_lb_urls])
+
+
+def fetch_haproxy_stats(haproxy_url):
+    try:
+        response = requests.get(haproxy_url + "/haproxy?stats;csv")
+        response.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        logger.exception("Caught exception when retrieving HAProxy"
+                         " stats from " + haproxy_url)
+        raise e
+
+    return response.text
+
+
+def fetch_combined_haproxy_stats(marathon_lb_urls):
+    raw = ''.join([ fetch_haproxy_stats(url) for url in marathon_lb_urls])
+    return parse_haproxy_stats(raw)
+
+
+def max_wait_exceeded(args, timestamp):
+    return (time.time() - timestamp > args.max_wait)
+
+
+def parse_haproxy_stats(csv_data):
+    rows = csv_data.splitlines()
+    headings = rows.pop(0).lstrip('# ').rstrip(',\n').split(',')
```

Comment:
Is there a reason that you aren't using python [`csv`](https://docs.python.org/3/library/csv.html) here?